### PR TITLE
Add auto_option for ImageMagick and add support for specifying a COMP…

### PR DESCRIPTION
**Pull request by cmcfarlen**
…ONENT to auto_option

### CODE REVIEW

**Review by bryancall — DISMISSED** [review]

File: cmake/AutoOptionHelpers.cmake [hunk 1]
```diff
@@ -37,18 +37,26 @@ function(_REGISTER_AUTO_OPTION _NAME _FEATURE_VAR _DESCRIPTION _DEFAULT)
...
   else()
     set(STRICTNESS REQUIRED)
   endif()
 
   foreach(PACKAGE_NAME ${_PACKAGE_DEPENDS})
-    find_package(${PACKAGE_NAME} ${STRICTNESS})
-    if(NOT ${PACKAGE_NAME}_FOUND)
-      set(${_FEATURE_VAR} FALSE)
-      break()
+    if(NOT ${_COMPONENT} STREQUAL OFF)
```
Comment (JosiahWI) [inline cmake/AutoOptionHelpers.cmake:48]: What about:
```suggestion
    if(${_COMPONENT})
```

`"OFF"`, `"0"`, `"FALSE"` and `""` should all have falsey values, I think.

File: cmake/AutoOptionHelpers.cmake [hunk 2]
```diff
@@ -37,18 +37,26 @@ function(_REGISTER_AUTO_OPTION _NAME _FEATURE_VAR _DESCRIPTION _DEFAULT)
   )
 endfunction()
 
-macro(_CHECK_PACKAGE_DEPENDS _OPTION_VAR _PACKAGE_DEPENDS _FEATURE_VAR)
+macro(_CHECK_PACKAGE_DEPENDS _OPTION_VAR _PACKAGE_DEPENDS _FEATURE_VAR _COMPONENT)
```
Comment (JosiahWI) [inline cmake/AutoOptionHelpers.cmake:40]: This would be more flexible as a list of components.

**Review by bryancall — DISMISSED** [review]

File: cmake/AutoOptionHelpers.cmake [hunk 3]
```diff
@@ -128,11 +140,17 @@ macro(auto_option _FEATURE_NAME)
     message(FATAL_ERROR "Invalid auto_option default ${ARG_DEFAULT}")
   endif()
 
+  if(ARG_COMPONENT)
+    set(COMPONENT ${ARG_COMPONENT})
+  else()
+    set(COMPONENT OFF)
```
Comment (JosiahWI) [inline cmake/AutoOptionHelpers.cmake:146]: I'm not sure about setting it to `OFF` here; it doesn't represent an option.

**Review by JosiahWI — CHANGES_REQUESTED** [review]

File: cmake/AutoOptionHelpers.cmake (hunk 1 above)
Comment (cmcfarlen) [reply cmake/AutoOptionHelpers.cmake:48]: I'm pretty sure I tried this and couldn't get it to work.

File: cmake/AutoOptionHelpers.cmake (hunk 2 above)
Comment (cmcfarlen) [reply cmake/AutoOptionHelpers.cmake:40]: agreed, but I couldn't get it to work providing two lists

File: cmake/AutoOptionHelpers.cmake (hunk 3 above)
Comment (cmcfarlen) [reply cmake/AutoOptionHelpers.cmake:146]: If I don't set it to off, the condition picking the COMPONENTS path would not work

**Comment by cmcfarlen** [comment]
[approve ci]

**Review by bryancall — APPROVED** [review]

**Pull request merged**